### ISSUE TITLE
NCCL errors when loading model with --gather_all_token_logits

### ISSUE THREAD

**Issue by dongluw** [triton-inference-server/tensorrtllm_backend#159]
I have a engine build with flags `--use_gpt_attention_plugin float16 --use_gemm_plugin float16 --enable_context_fmha --alibi True --world_size 4 --use_parallel_embedding --gather_all_token_logits`, I was able to run it using `run.py`, however when I tried to load it with Triton (Python runtime) I got the following errors.

It worked when I didn't build it with the `--gather_all_token_logits` option.
I added the `gather_all_token_logits` ModelConfig to model.py [here](https://github.com/triton-inference-server/tensorrtllm_backend/blob/release/0.5.0/all_models/gpt/tensorrt_llm/1/model.py#L98)

```
I1123 17:47:59.774343 46608 model_lifecycle.cc:819] successfully loaded 'preprocessing'
I1123 17:47:59.822943 46610 model_lifecycle.cc:819] successfully loaded 'preprocessing'
root@tif-a100-us-east1:/tensorrtllm_backend# [TensorRT-LLM][ERROR] Failed, NCCL error /app/tensorrt_llm/cpp/tensorrt_llm/runtime/ncclCommunicator.cpp:122 'internal error - please report this issue to the NCCL developers'

[TensorRT-LLM][ERROR] Failed, NCCL error /app/tensorrt_llm/cpp/tensorrt_llm/runtime/ncclCommunicator.cpp:122 'internal error - please report this issue to the NCCL developers'

double free or corruption (fasttop)
[tif-a100-us-east1:46796] *** Process received signal ***
[tif-a100-us-east1:46796] Signal: Aborted (6)
[tif-a100-us-east1:46796] Signal code:  (-6)
double free or corruption (fasttop)
[tif-a100-us-east1:46968] *** Process received signal ***
[tif-a100-us-east1:46968] Signal: Aborted (6)
[tif-a100-us-east1:46968] Signal code:  (-6)
[tif-a100-us-east1:46796] [ 0] [tif-a100-us-east1:46968] [ 0] /usr/lib/x86_64-linux-gnu/libc.so.6(+0x42520)[0x7fdc42617520]
[tif-a100-us-east1:46968] [ 1] /usr/lib/x86_64-linux-gnu/libc.so.6(+0x42520)[0x7f418d617520]
[tif-a100-us-east1:46796] [ 1] /usr/lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x12c)[0x7f418d66ba7c]
[tif-a100-us-east1:46796] [ 2] /usr/lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x12c)[0x7fdc4266ba7c]
[tif-a100-us-east1:46968] [ 2] /usr/lib/x86_64-linux-gnu/libc.so.6(raise+0x16)[0x7f418d617476]
[tif-a100-us-east1:46796] [ 3] /usr/lib/x86_64-linux-gnu/libc.so.6(raise+0x16)[0x7fdc42617476]
[tif-a100-us-east1:46968] [ 3] /usr/lib/x86_64-linux-gnu/libc.so.6(abort+0xd3)[0x7f418d5fd7f3]
[tif-a100-us-east1:46796] [ 4] /usr/lib/x86_64-linux-gnu/libc.so.6(abort+0xd3)[0x7fdc425fd7f3]
[tif-a100-us-east1:46968] [ 4] /usr/lib/x86_64-linux-gnu/libc.so.6(+0x896f6)[0x7f418d65e6f6]
[tif-a100-us-east1:46796] [ 5] /usr/lib/x86_64-linux-gnu/libc.so.6(+0x896f6)[0x7fdc4265e6f6]
[tif-a100-us-east1:46968] [ 5] /usr/lib/x86_64-linux-gnu/libc.so.6(+0xa0d7c)[0x7fdc42675d7c]
[tif-a100-us-east1:46968] [ 6] /usr/lib/x86_64-linux-gnu/libc.so.6(+0xa2a4a)[0x7fdc42677a4a]
[tif-a100-us-east1:46968] [ 7] /usr/lib/x86_64-linux-gnu/libc.so.6(+0xa0d7c)[0x7f418d675d7c]
[tif-a100-us-east1:46796] [ 6] /usr/lib/x86_64-linux-gnu/libc.so.6(+0xa2a4a)[0x7f418d677a4a]
[tif-a100-us-east1:46796] [ 7] /usr/lib/x86_64-linux-gnu/libc.so.6(free+0x73)[0x7f418d67a4d3]
/usr/lib/x86_64-linux-gnu/libc.so.6(free+0x73)[0x7fdc4267a4d3]
[tif-a100-us-east1:46968] [ 8] /usr/lib/x86_64-linux-gnu/libc.so.6(+0x45495)[0x7fdc4261a495]
[tif-a100-us-east1:46968] [tif-a100-us-east1:46796] [ 8] [ 9] /usr/lib/x86_64-linux-gnu/libc.so.6(on_exit+0x0)[0x7fdc4261a610]
[tif-a100-us-east1:46968] /usr/lib/x86_64-linux-gnu/libc.so.6(+0x45495)[0x7f418d61a495]
[tif-a100-us-east1:46796] [ 9] [10] /usr/lib/x86_64-linux-gnu/libc.so.6(on_exit+0x0)[0x7f418d61a610]
[tif-a100-us-east1:46796] [10] /app/tensorrt_llm/tensorrt_llm/libs/libth_common.so(_ZN12tensorrt_llm7runtime16NcclCommunicator18createPipelineCommERKNS0_11WorldConfigERN8nvinfer17ILoggerE+0x280)[0x7fd9db981140]
[tif-a100-us-east1:46968] [11] /app/tensorrt_llm/tensorrt_llm/libs/libth_common.so(_ZN12tensorrt_llm7runtime16NcclCommunicator18createPipelineCommERKNS0_11WorldConfigERN8nvinfer17ILoggerE+0x280)[0x7f3f27981140]
[tif-a100-us-east1:46796] [11] /app/tensorrt_llm/tensorrt_llm/libs/libth_common.so(_ZN9torch_ext18NcclCommunicatorOpC1Elll+0xc0)[0x7fd9db8ea590]
[tif-a100-us-east1:46968] [12] /app/tensorrt_llm/tensorrt_llm/libs/libth_common.so(_ZN9torch_ext18NcclCommunicatorOpC1Elll+0xc0)[0x7f3f278ea590]
[tif-a100-us-east1:46796] [12] /app/tensorrt_llm/tensorrt_llm/libs/libth_common.so(_ZNSt17_Function_handlerIFvRSt6vectorIN3c106IValueESaIS2_EEEZN5torch6class_IN9torch_ext18NcclCommunicatorOpEE12defineMethodIZNSB_3defIJlllEEERSB_NS7_6detail5typesIvJDpT_EEESsSt16initializer_listINS7_3argEEEUlNS1_14tagged_capsuleISA_EElllE_EEPNS7_3jit8FunctionESsT_SsSM_EUlS5_E_E9_M_invokeERKSt9_Any_dataS5_+0x9d)[0x7fd9db8eb97d]
[tif-a100-us-east1:46968] [13] /app/tensorrt_llm/tensorrt_llm/libs/libth_common.so(_ZNSt17_Function_handlerIFvRSt6vectorIN3c106IValueESaIS2_EEEZN5torch6class_IN9torch_ext18NcclCommunicatorOpEE12defineMethodIZNSB_3defIJlllEEERSB_NS7_6detail5typesIvJDpT_EEESsSt16initializer_listINS7_3argEEEUlNS1_14tagged_capsuleISA_EElllE_EEPNS7_3jit8FunctionESsT_SsSM_EUlS5_E_E9_M_invokeERKSt9_Any_dataS5_+0x9d)[0x7f3f278eb97d]
[tif-a100-us-east1:46796] [13] /usr/local/lib/python3.10/dist-packages/torch/lib/libtorch_python.so(+0x93f4be)[0x7fdc38f6e4be]
[tif-a100-us-east1:46968] [14] /usr/local/lib/python3.10/dist-packages/torch/lib/libtorch_python.so(+0x93f4be)[0x7f4142f2f4be]
[tif-a100-us-east1:46796] [14] /usr/local/lib/python3.10/dist-packages/torch/lib/libtorch_python.so(+0x93ca5e)[0x7f4142f2ca5e]
[tif-a100-us-east1:46796] [15] /usr/local/lib/python3.10/dist-packages/torch/lib/libtorch_python.so(+0x93ca5e)[0x7fdc38f6ba5e]
[tif-a100-us-east1:46968] [15] /usr/local/lib/python3.10/dist-packages/torch/lib/libtorch_python.so(+0x93eaf9)[0x7f4142f2eaf9]
[tif-a100-us-east1:46796] [16] /usr/local/lib/python3.10/dist-packages/torch/lib/libtorch_python.so(+0x93eaf9)[0x7fdc38f6daf9]
[tif-a100-us-east1:46968] [16] /usr/local/lib/python3.10/dist-packages/torch/lib/libtorch_python.so(+0x3eeac4)[0x7fdc38a1dac4]
[tif-a100-us-east1:46968] [17] /usr/local/lib/python3.10/dist-packages/torch/lib/libtorch_python.so(+0x3eeac4)[0x7f41429deac4]
[tif-a100-us-east1:46796] [17] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x128023)[0x7f418db51023]
[tif-a100-us-east1:46796] [18] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x128023)[0x7fdc42b51023]
[tif-a100-us-east1:46968] [18] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x8c)[0x7fdc42b08adc]
[tif-a100-us-east1:46968] [19] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x8c)[0x7f418db08adc]
[tif-a100-us-east1:46796] [19] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0xe243a)[0x7fdc42b0b43a]
[tif-a100-us-east1:46968] [20] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0xe243a)[0x7f418db0b43a]
[tif-a100-us-east1:46796] [20] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x14ea6b)[0x7fdc42b77a6b]
[tif-a100-us-east1:46968] [21] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x14ea6b)[0x7f418db77a6b]
[tif-a100-us-east1:46796] [21] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x8c)[0x7fdc42b08adc]
[tif-a100-us-east1:46968] [22] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x8c)[0x7f418db08adc]
[tif-a100-us-east1:46796] [22] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0x9dbc)[0x7fdc42aa4a1c]
[tif-a100-us-east1:46968] [23] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0x9dbc)[0x7f418daa4a1c]
[tif-a100-us-east1:46796] [23] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x1c23af)[0x7fdc42beb3af]
[tif-a100-us-east1:46968] [24] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x1c23af)[0x7f418dbeb3af]
[tif-a100-us-east1:46796] [24] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_FastCallDictTstate+0x57)[0x7fdc42b0abf7]
[tif-a100-us-east1:46968] [25] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_FastCallDictTstate+0x57)[0x7f418db0abf7]
[tif-a100-us-east1:46796] [25] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_Call_Prepend+0xe0)[0x7f418db0ae10]
[tif-a100-us-east1:46796] [26] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_Call_Prepend+0xe0)[0x7fdc42b0ae10]
[tif-a100-us-east1:46968] [26] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x14f0fe)[0x7fdc42b780fe]
[tif-a100-us-east1:46968] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x14f0fe)[0x7f418db780fe]
[tif-a100-us-east1:46796] [27] [27] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x14388e)[0x7fdc42b6c88e]
[tif-a100-us-east1:46968] [28] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x8c)[0x7fdc42b08adc]
[tif-a100-us-east1:46968] [29] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(+0x14388e)[0x7f418db6c88e]
[tif-a100-us-east1:46796] [28] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x8c)[0x7f418db08adc]
[tif-a100-us-east1:46796] [29] /usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0xa62a)[0x7f418daa528a]
[tif-a100-us-east1:46796] *** End of error message ***
/usr/lib/x86_64-linux-gnu/libpython3.10.so.1.0(_PyEval_EvalFrameDefault+0xa62a)[0x7fdc42aa528a]
[tif-a100-us-east1:46968] *** End of error message ***
E1123 17:48:38.752626 46609 backend_model.cc:553] ERROR: Failed to create instance: Stub process 'tensorrt_llm_0_0' is not healthy.
E1123 17:48:38.753885 46609 model_lifecycle.cc:622] failed to load 'tensorrt_llm' version 1: Internal: Stub process 'tensorrt_llm_0_0' is not healthy.
I1123 17:48:38.754129 46609 model_lifecycle.cc:757] failed to load 'tensorrt_llm'
E1123 17:48:38.754410 46609 model_repository_manager.cc:563] Invalid argument: ensemble 'ensemble' depends on 'tensorrt_llm' which has no loaded version. Model 'tensorrt_llm' loading failed with error: version 1 is at UNAVAILABLE state: Internal: Stub process 'tensorrt_llm_0_0' is not healthy.;
I1123 17:48:38.754893 46609 server.cc:604] 
+------------------+------+
| Repository Agent | Path |
+------------------+------+
+------------------+------+

I1123 17:48:38.755007 46609 server.cc:631] 
+---------+---------------------------------------------------------+------------------------------------------------------------------+
| Backend | Path                                                    | Config                                                           |
+---------+---------------------------------------------------------+------------------------------------------------------------------+
| pytorch | /opt/tritonserver/backends/pytorch/libtriton_pytorch.so | {}                                                               |
| python  | /opt/tritonserver/backends/python/libtriton_python.so   | {"cmdline":{"auto-complete-config":"false","backend-directory":" |
|         |                                                         | /opt/tritonserver/backends","min-compute-capability":"6.000000", |
|         |                                                         | "shm-region-prefix-name":"prefix1_","default-max-batch-size":"4" |
|         |                                                         | }}                                                               |
+---------+---------------------------------------------------------+------------------------------------------------------------------+

I1123 17:48:38.755089 46609 server.cc:674] 
+----------------+---------+------------------------------------------------------------------------+
| Model          | Version | Status                                                                 |
+----------------+---------+------------------------------------------------------------------------+
| postprocessing | 1       | READY                                                                  |
| preprocessing  | 1       | READY                                                                  |
| tensorrt_llm   | 1       | UNAVAILABLE: Internal: Stub process 'tensorrt_llm_0_0' is not healthy. |
+----------------+---------+------------------------------------------------------------------------+

```

**Comment by dongluw** [comment]
Hi @byshiue, thanks for the reply, wondering what's the timeline for supporting `gather_all_token_logits` in trt-llm backend?

**Comment by byshiue** [comment]
It is not determiend yet.

**Comment by byshiue** [comment]
This feature is supported and the document is [here](https://github.com/triton-inference-server/tensorrtllm_backend/tree/main?tab=readme-ov-file#return-context-logits-andor-generation-logits).

Close this bug. Feel free to ask here again if you encounter any issue.
